Mathlib/MeasureTheory/Measure/Haar/Unique.lean

```diff
@@ -282,56 +302,466 @@ lemma integral_isMulLeftInvariant_eq_smul_of_hasCompactSupport
+lemma haarScalarFactor_pos_of_isHaarMeasure (μ' μ : Measure G) [IsHaarMeasure μ]
+    [IsHaarMeasure μ'] : 0 < haarScalarFactor μ' μ :=
+  pos_iff_ne_zero.2 (fun H ↦ by simpa [H] using haarScalarFactor_eq_mul μ' μ μ')
+
+@[deprecated] alias haarScalarFactor_pos_of_isOpenPosMeasure :=
+  haarScalarFactor_pos_of_isHaarMeasure  -- deprecated on 2024-02-12
+
+@[deprecated] alias addHaarScalarFactor_pos_of_isOpenPosMeasure :=
+  addHaarScalarFactor_pos_of_isAddHaarMeasure  -- deprecated on 2024-02-12
+
+/-!
+### Uniqueness of measure of sets with compact closure
+
+Two left invariant measures give the same measure to sets with compact closure, up to the
+scalar `haarScalarFactor μ' μ`.
+
+This is a tricky argument, typically not done in textbooks (the textbooks version all require one
+version of regularity or another). Here is a sketch, based on
+McQuillan's answer at https://mathoverflow.net/questions/456670/.
+
+Assume for simplicity that all measures are normalized, so that the scalar factors are all `1`.
+First, from the fact that `μ` and `μ'` integrate in the same way compactly supported functions,
+they give the same measure to compact "zero sets", i.e., sets of the form `f⁻¹ {1}`
+for `f` continuous and compactly supported.
+See `measure_preimage_isMulLeftInvariant_eq_smul_of_hasCompactSupport`.
+
+If `μ` is inner regular, a theorem of Halmos shows that any measurable set `s` of finite measure can
+be approximated from inside by a compact zero set `k`. Then `μ s ≤ μ k + ε = μ' k + ε ≤ μ' s + ε`.
+Letting `ε` tend to zero, one gets `μ s ≤ μ' s`.
+See `smul_measure_isMulInvariant_le_of_isCompact_closure`.
+
+Assume now that `s` is a measurable set of compact closure. It is contained in a compact
+zero set `t`. The same argument applied to `t - s` gives `μ (t \ s) ≤ μ' (t \ s)`, i.e.,
+`μ t - μ s ≤ μ' t - μ' s`. As `μ t = μ' t` (since these are zero sets), we get the inequality
+`μ' s ≤ μ s`. Together with the previous one, this gives `μ' s = μ s`.
+See `measure_isMulInvariant_eq_smul_of_isCompact_closure_of_innerRegularCompactLTTop`.
+
+If neither `μ` nor `μ'` is inner regular, we can use the existence of another inner regular
+left-invariant measure `ν`, so get `μ s = ν s = μ' s`, by applying twice the previous argument.
+Here, the uniqueness argument uses the existence of a Haar measure with a nice behavior!
+See `measure_isMulInvariant_eq_smul_of_isCompact_closure_of_measurableSet`.
+
+Finally, if `s` has compact closure but is not measurable, its measure is the infimum of the
+measures of its measurable supersets, and even of those contained in `closure s`. As `μ`
+and `μ'` coincide on these supersets, this yields `μ s = μ' s`.
+See `measure_isMulInvariant_eq_smul_of_isCompact_closure`.
+-/
+
+/-- Two left invariant measures give the same mass to level sets of continuous compactly supported
+functions, up to the scalar `haarScalarFactor μ' μ`.
+Superseded by `measure_isMulInvariant_eq_smul_of_isCompact_closure`, which works for any set with
```

Comment:
All the lemmas whose docstring mentions "superseded" are intermediate steps in the proof of the main theorem below, `measure_isMulInvariant_eq_smul_of_isCompact_closure`. So I can't remove them. The role of this docstring is to explain that the user shouldn't use them because there is a better version available below.

Reply:
I have expanded the docstrings to clarify that these are intermediate lemmas.